### PR TITLE
Create str-join.md

### PR DESCRIPTION
**Pull request by SavagePastaMan**
Create a tag to showcase `str.join`

Here is a screenshot of the tag 
image

### CODE REVIEW

**Comment by Akarys42** [comment]
Hey @SavagePastaMan, can you please add a bit more content to the PR body? right now it isn't saying much about the PR either how it looks inside the server. Also, don't forget to lint!

**Comment by SavagePastaMan** [comment]
i'm just using the github editor :pensive:. I also don't have a local instance of the bot so I can't get screenshots 😬.

**Review by ChrisLovering — APPROVED** [review]

File: bot/resources/tags/str-join.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+**Joining Iterables**
+
+Suppose you want to nicely display a list (or some other iterable). The naive solution would be something like this.
```
Comment (mbaruh) [inline bot/resources/tags/str-join.md:3]:
```suggestion
If you want to display a list (or some other iterable), you can write:
```

File: bot/resources/tags/str-join.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+```py
+colors = ['red', 'green', 'blue', 'yellow']
+output = ""
+separator = ", "
+for color in colors:
+    output += color + separator
+print(output)
+# Prints 'red, green, blue, yellow, '
+```
+However, this solution is flawed. The separator is still added to the last element, and it is slow.
```
Comment (mbaruh) [inline bot/resources/tags/str-join.md:13]:
```suggestion
However, the separator is still added to the last element, and it is relatively slow.
```

File: bot/resources/tags/str-join.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+The better solution is to use `str.join`.
```
Comment (mbaruh) [inline bot/resources/tags/str-join.md:15]:
```suggestion
A better solution is to use `str.join`.
```

File: bot/resources/tags/str-join.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+```py
+colors = ['red', 'green', 'blue', 'yellow']
+separator = ", "
+print(separator.join(colors))
+# Prints 'red, green, blue, yellow'
+```
+This solution is much simpler, faster, and solves the problem of the extra separator. An important thing to note is that you can only `str.join` strings. For a list of ints,
```
Comment (mbaruh) [inline bot/resources/tags/str-join.md:22]:
```suggestion
An important thing to note is that you can only `str.join` strings. For a list of ints,
```

**Review by mbaruh — CHANGES_REQUESTED** [review]
I'd keep it as short as possible, and also maybe a bit less opinionated.

**Comment by Xithrius** [comment]
Author will not be able to complete this PR, so I'll be taking over.

**Review by mbaruh — APPROVED** [review]

**Review by Akarys42 — APPROVED** [review]

**Pull request merged**